x/quarantine/keeper/grpc_query_test.go

```diff
@@ -0,0 +1,553 @@
+package keeper_test
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/types/address"
+	"github.com/cosmos/cosmos-sdk/types/query"
+
+	"github.com/provenance-io/provenance/x/quarantine"
+	"github.com/provenance-io/provenance/x/quarantine/testutil"
+)
+
+// These tests are initiated by TestKeeperTestSuite in keeper_test.go
+
+func (s *TestSuite) TestIsQuarantined() {
+	addrQuarantinedAcc := testutil.MakeTestAddr("iq", 0)
+	addrQuarantinedStr := addrQuarantinedAcc.String()
+	addrNormalStr := testutil.MakeTestAddr("iq", 1).String()
+
+	s.Require().NoError(s.keeper.SetOptIn(s.sdkCtx, addrQuarantinedAcc), "SetOptIn")
+
+	tests := []struct {
+		name string
+		req  *quarantine.QueryIsQuarantinedRequest
+		resp *quarantine.QueryIsQuarantinedResponse
+		err  []string
+	}{
+		{
+			name: "nil req",
+			req:  nil,
+			err:  []string{"empty request"},
+		},
+		{
+			name: "empty to address",
+			req: &quarantine.QueryIsQuarantinedRequest{
+				ToAddress: "",
+			},
+			err: []string{"to address cannot be empty"},
+		},
+		{
+			name: "bad to address",
+			req: &quarantine.QueryIsQuarantinedRequest{
+				ToAddress: "yupitisbad",
+			},
+			err: []string{"invalid to address", "decoding bech32 failed"},
+		},
+		{
+			name: "quarantined address",
+			req: &quarantine.QueryIsQuarantinedRequest{
+				ToAddress: addrQuarantinedStr,
+			},
+			resp: &quarantine.QueryIsQuarantinedResponse{
+				IsQuarantined: true,
+			},
+		},
+		{
+			name: "normal address",
+			req: &quarantine.QueryIsQuarantinedRequest{
+				ToAddress: addrNormalStr,
+			},
+			resp: &quarantine.QueryIsQuarantinedResponse{
+				IsQuarantined: false,
+			},
+		},
+	}
+
+	for _, tc := range tests {
+		s.Run(tc.name, func() {
+			resp, err := s.keeper.IsQuarantined(s.stdlibCtx, tc.req)
+			if s.AssertErrorContents(err, tc.err, "IsQuarantined error") {
+				s.Assert().Equal(tc.resp, resp, "IsQuarantined result")
+			}
+		})
+	}
+}
```

Comment:
Here again, `s.AssertErrorContents` cannot be replaced with `s.Require().NoError` because they test two different things. I also don't like using `Require()` unless there is absolutely nothing left that might possibly be checked if the require fails.

In this case, I don't use a require check so that later, if there's more things to check, those checks can be added without the updater needing to go back and change a require to an assert.